acme/authority.go

```diff
@@ -54,8 +57,24 @@ var (
 	orderTable             = []byte("acme_orders")
 	ordersByAccountIDTable = []byte("acme_account_orders_index")
 	certTable              = []byte("acme_certs")
+	ordinal                int
 )
 
+// Ordinal is used during challenge retries to indicate ownership.
+func init() {
+	ordstr := os.Getenv("STEP_CA_ORDINAL")
```

Comment:
Leaving myself a note to discuss the env var name. Are we using `STEP_CA_xxx` in other places? I know that we use `STEPxxx`.

Reply:
In the `step` CLI, each flag has a special environment variable that we can set up, for example, if the flag is `--ca-url` you can set up `STEP_CA_URL`, in the special case where the flag has an alias, only the first alias is taken into account `--force, -f` will be only `STEP_FORCE`.

So if we have the flag `--ca-ordinal`, that we don't afaik, they will collapse.

Reply:
We could add this all the way into the UI and make it a flag. My gut impression was that this wouldn't be an option that people use frequently but if everything is configurable via a flag then I can add that.

Reply:
My comment was about how `step` works, I was not suggesting to add a flag, yet.

After reviewing everything, I don't like this as an environment variable, I'm going to propose a couple of solutions:
1. Add a flag to `step-ca` and pass it here, we need to be careful to not break Caddy.
2. Read a random string from the db, if it's not there, create it and use that one.

Reply:
I'm currently working on adding a flag.